pyvista/plotting/colors.py

```diff
@@ -886,15 +884,18 @@ def __hash__(self):  # pragma: no cover
 
     def __getitem__(self, item):
         """Support indexing the float RGBA representation for backward compatibility."""
-        if isinstance(item, (str, int, np.integer)):
+        if not isinstance(item, (str, int, np.integer)):
+            raise TypeError("Invalid index specified, only strings and integers are supported.")
+        if isinstance(item, str):
             for i, cnames in enumerate(self.CHANNEL_NAMES):
                 if item in cnames:
                     item = i
-            if item < 0 or item > 3:
-                raise IndexError("Colors have only 4 channels.")
-            return self.float_rgba[item]
-        else:
-            raise TypeError("Invalid index specified, only strings and integers are supported.")
+                    break
+            else:
+                raise ValueError(f"Invalid string index {item!r}.")
+        if not 0 <= item < 4:
```

Comment:
I wasn't going to suggest this because this vs the original are close enough that going out of our way to change it wouldn't be reasonable, but since I was already touching multiple lines here I figured I'd switch the test here too. If you think this is harder to parse feel free to revert (I tend to find contiguous ranges easier to reason about). `item <= 3` on the right side is also a possible alternative.

Reply:
Ok for me. Didn't know `for/else` was a thing, looks interesting/peculiar at the same time but fits the goal.
I will add some tests for the raised exceptions later to satisfy codecov.